lib/dev_repo.mli

```diff
@@ -5,9 +5,11 @@ type t
 val from_string : string -> t
 val to_string : t -> string
 
-val repo_name : t -> string
+val repo_name : t -> (string, Rresult.R.msg) result
 (** Returns the name of the repo given the dev-repo.
     E.g. [repo_name (from_string "https://github.com/ocamllabs/opam-monorepo.git")]
-    returns ["opam-monorepo"]. *)
+    returns ["opam-monorepo"].
```

Comment:
Can you describe the "high-level" approach it does here, e.g. determining the last component and removing everything before the dot? I think having a english language specification on how the repo name is determined would be useful.